src/north/amazon/AmazonS3.class.spec.js

```diff
@@ -0,0 +1,50 @@
+const fs = require('fs')
+const path = require('path')
+const AWS = require('aws-sdk')
+const ApiHandler = require('../ApiHandler.class')
+const AmazonS3 = require('./AmazonS3.class')
+const config = require('../../../tests/testConfig').default
+
+// Mock database service
+jest.mock('../../services/database.service', () => {})
+
+// Mock logger
+jest.mock('../../engine/Logger.class')
+
+// Mock engine
+const engine = jest.genMockFromModule('../../engine/Engine.class')
+engine.configService = { getConfig: () => ({ engineConfig: config.engine }) }
+
+beforeEach(() => {
+  jest.resetAllMocks()
+})
+
+describe('Amazone S3 north', () => {
+  const AmazonS3Config = config.north.applications[8]
+  const AmazonS3North = new AmazonS3(AmazonS3Config, engine)
+
+  it('should be properly initialized', () => {
+    expect(AmazonS3North.bucket).toEqual(AmazonS3Config.AmazonS3.bucket)
+    expect(AmazonS3North.folder).toEqual(AmazonS3Config.AmazonS3.folder)
+    AWS.config.update = jest.fn()
+    const accessKeyId = '' // doubt about what to put here
+    const secretAccessKey = ''
+    expect(AWS.config.update).toHaveBeenCalledWith(accessKeyId, secretAccessKey)
+    expect(AmazonS3North.s3).toHaveBeenCalledTimes(1)
+    expect(AmazonS3North.canHandleFiles).toBeTruthy()
+  })
+
+  it('should properly handle file', async () => {
+    const filePath = '/path/to/file/example.file'
+    jest.spyOn(fs, 'createReadStream').mockImplementation(() => ({ key: '' })) // necessary ?
+    const expectedparams = {
+      Bucket: AmazonS3Config.AmazonS3.bucket,
+      Body: AmazonS3Config.AmazonS3.body,
+      Key: `${AmazonS3Config.AmazonS3.folder}/${AmazonS3Config.AmazonS3.getFilenameWithoutTimestamp(filePath)}`,
+      // doubt about the last parameter
+    }
```

Comment:
Bucket comes from the config file, so it is correct, but Body is something else. Try to understand the upload method to understand what is the Body param.
Same problem with the key : there is no method in the config object. Ask yourself what is AmazonS3Config, what's its value when it is set, etc (you can console log its value if it can help you)